src/library.js

```diff
@@ -397,21 +397,6 @@ LibraryManager.library = {
     return ((Date.now() - _clock.start) * ({{{ cDefine('CLOCKS_PER_SEC') }}} / 1000))|0;
   },
 
-  time__sig: 'ii',
-  time: function(ptr) {
-    {{{ from64('ptr') }}};
-    var ret = (Date.now()/1000)|0;
-    if (ptr) {
-      {{{ makeSetValue('ptr', 0, 'ret', 'i32') }}};
-    }
-    return ret;
-  },
-
-  difftime__sig: 'dii',
-  difftime: function(time1, time0) {
-    return time1 - time0;
-  },
```

Comment:
Perhaps we can land the moving of these two functions to native code separately, before landing this PR.

The `difftime` change looks like a no-brainer.. the `time` change probably is too, but could effect code size.

Reply:
Yes. The `time` change is responsible for the 2/3% code size regression on the `test-other` test suite.

I am very tempted to remove most of the date stuff from the JS side, and just have a single `_time_js` function which just returns Date.now() as a double.

This should help with bundle size and performance, because we won't have to pass i64s on the wasm/js boundary anymore

Reply:
Yup, this was what I was worried about with increasing the width of time_t.   Any refactors that move stuff into native code without regressing size sounds good to me.

Can I recommend a PR that just moves time() and timediff() first.. rather than trying to make one big refactor.

Reply:
Yes, I am going to do that

Reply:
Ah, sorry for the (big) delay in my response.
I was working on something very similar, but didn't have time to finish
This is great! I will continue my work on this PR asap